jwst/ami/tests/test_ami_analyze.py

```diff
@@ -592,29 +594,201 @@ def test_hexee_glimit():
+           5.75956679-0.28527278j, 4.43309812-0.17169358j, 2.16095243-0.00986439j],
+         [ 0.85814396-0.29934852j, 2.95391668-0.74894583j, 4.39936028-0.9414124j,
+           4.66263759-0.840202j,   3.6397516 -0.48892695j, 1.69794671+0.00647795j],
+         [-0.53224117-0.7665882j,  1.09507456-1.3630922j,  2.35075667-1.55901736j,
+           2.7532696 -1.32158428j, 2.14239062-0.74031707j, 0.74611461+0.00885311j]] )
+
+    assert_allclose(asf, true_asf, atol=1E-8 )
+
+
+def test_analyticnrm2_interf():
+    ''' Test of interf() in the analyticnrm2 module '''
+
+    ASIZE = 4
+    kx = np.arange(ASIZE * ASIZE).reshape((ASIZE, ASIZE))
+    ky = np.arange(ASIZE * ASIZE).reshape((ASIZE, ASIZE))
+    vv = np.arange(ASIZE)
+
+    for ii in np.arange(ASIZE):
+        kx[:,ii] = vv
+        ky[ii,:] = vv
+
+    # Clean up any attributes that may have been added earlier
+    for kk in list( (interf.__dict__).keys()):
+        delattr( interf, kk )
+
+    pixel, fov, oversample, ctrs, d, lam, phi, centering = setup_SF()
+
+    interf.lam = lam
+    interf.offx = 0.5
+    interf.offy = 0.5
+    interf.ctrs = ctrs
+    interf.d = d
+    interf.phi = phi
+    interf.pitch = pixel / float(oversample)
+
+    interference = interf(kx, ky)
+
+    true_interference = np.array(
+       [[6.65604548+0.32967559j, 6.55020282-0.35898074j,
+         5.10088264-1.09153011j, 2.74363797-1.81957549j ],
+        [6.55020282+0.35898074j, 6.65604548-0.32967559j,
+         5.40614218-0.97418068j, 3.21342278-1.5424603j ],
+        [4.86319369+0.26557752j, 5.14000033-0.19907186j,
+         4.23407392-0.56876212j, 2.50871764-0.86690377j ],
+        [2.18032216+0.05966983j, 2.52211181-0.01151302j,
+         1.98827833+0.00758561j, 0.87949091+0.01043571j]]
+        )
+
+    assert_allclose(interference, true_interference, atol=1E-8 )
+
+
+def test_analyticnrm2_phasor():
+    ''' Test of phasor() in the analyticnrm2 module '''
+
+    ASIZE = 4
+    kx = np.arange(ASIZE * ASIZE).reshape((ASIZE, ASIZE))
+
+    for ii in np.arange(ASIZE):
+        kx[:,ii] = ii
+
+    ky = kx.transpose()
+
+    hx = 0.06864653345335156
+    hy = -2.6391073592116028
+
+    lam = 2.3965000082171173e-06
+    phi = 0.0
+    pitch = 1.0375012775744072e-07
+
+    result = phasor( kx, ky, hx, hy, lam, phi, pitch )
+
+    true_result = np.array(
+      [[ 1. -0.j, 0.99982567-0.01867173j, 0.99930273-0.03733694j,
+         0.99843138-0.05598914j ],
+       [ 0.75320598+0.65778473j, 0.76535665+0.6436064j, 0.77724047+0.62920367j,
+         0.78885329+0.61458156j ],
+       [ 0.1346385+0.99089478j, 0.15311675+0.98820811j, 0.1715416+0.98517688j,
+         0.18990665+0.98180215j ],
+       [-0.55038493+0.83491103j, -0.53469975+0.84504211j, -0.51882814+0.85487856j,
+        -0.50277564+0.86441695j ]]
+      )
+
+    assert_allclose( result, true_result, atol=1E-8 )
+
+
+def setup_SF():
+    ''' Initialize values for these parameters needed for the analyticnrm2 tests:
+
+        pixel (optional, via **kwargs) : float
```

Comment:
Is the "(optional, via **kwargs)" a cut & paste error?
You probably ought to include a "Returns" label and underline to make it clear that these are output values.